nimare/correct/correct.py

```diff
@@ -0,0 +1,138 @@
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+import numpy as np
+import statsmodels.stats.multitest as mc
+
+from ..base import MetaResult
+from ..stats import p_to_z
+
+
+class Corrector(metaclass=ABCMeta):
+    '''
+    Base class for multiple comparison correction methods.
+    '''
+
+    # The name of the method that must be implemented in an Estimator class
+    # in order to override the default correction method.
+    _correction_method = None
+
+    # Maps that must be available in the MetaResult instance
+    _required_maps = ('p',)
+
+    def __init__(self):
+        pass
+
+    @abstractproperty
+    def _name_suffix(self):
+        pass
+
+    def _validate_input(self, result):
+        if not isinstance(result, MetaResult):
+            raise ValueError("First argument to transform() must be an "
+                             "instance of class MetaResult, not {}."
+                             .format(type(result)))
+        for rm in self._required_maps:
+            if not result.maps.get(rm):
+                raise ValueError("{0} requires {1} maps to be present in the "
+                                 "MetaResult, but none were found."
+                                 .format(type(self), rm))
+
+    def _generate_secondary_maps(self, result, corr_maps):
+        # Generates corrected version of z and log-p maps if they exist
+        p = corr_maps['p']
+        if 'z' in result.maps:
+            corr_maps.maps['z'] = p_to_z(p) * np.sign(result.maps['z'])
+        if 'log_p' in result.maps:
+            corr_maps['log_p'] = -np.log10(p)
+        return corr_maps
+
+    def transform(self, result, **kwargs):
+        est = result.estimator
+        method = self._correction_method
+
+        # Make sure we return a copy of the MetaResult
+        result = result.copy()
+
+        # If a correction method with the same name exists in the current
+        # MetaEstimator, use it. Otherwise fall back on _transform.
+        if (method is not None and hasattr(est, method)):
+            corr_maps = getattr(est, method)(result, **kwargs)
+        else:
+            self._validate_input(result)
+            corr_maps = self._transform(result)
+
+        # Update corrected map names and add them to maps dict
+        corr_maps = {(k + self._name_suffix): v for k, v in corr_maps.items()}
+        result.maps.update(corr_maps)
```

Comment:
See earlier note. We're updating the `.maps` dict, so it definitely needs to be explicitly copied inside `MetaResults.copy()`, otherwise both copies of the `MetaResults` instance will have the same (modified) `maps` dict.